staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -111,13 +111,88 @@ func getItemsPtr(list runtime.Object) (interface{}, error) {
 	}
 }
 
+// ExtractList returns obj's Items element as an array of runtime.Objects.
+// Returns an error if obj is not a List type (does not have an Items member).
+func ExtractList(obj runtime.Object) ([]runtime.Object, error) {
+	return extractList(obj, false)
+}
+
+// ExtractListWithAlloc returns obj's Items element as an array of runtime.Objects.
+// the difference from ExtractList is that an object of the same type is regenerated and assigned a value,
+// so that the returned Object list and obj have independent life cycles.
+// Returns an error if obj is not a List type (does not have an Items member).
+func ExtractListWithAlloc(obj runtime.Object) ([]runtime.Object, error) {
+	return extractList(obj, true)
+}
+
 // EachListItem invokes fn on each runtime.Object in the list. Any error immediately terminates
 // the loop.
 func EachListItem(obj runtime.Object, fn func(runtime.Object) error) error {
+	return eachListItem(obj, false, fn)
+}
+
+// EachListItemWithAlloc invokes fn on each runtime.Object in the list.
+// the difference from  EachListItem is that an object of the same type is regenerated and assigned a value,
+// so that the returned Object list and obj have independent life cycles.
+func EachListItemWithAlloc(obj runtime.Object, fn func(runtime.Object) error) error {
+	return eachListItem(obj, true, fn)
+}
+
+// extractList A helper function that extract items in the list object according to the allocNew strategy.
+func extractList(obj runtime.Object, alloc bool) ([]runtime.Object, error) {
```

Comment:
moving this around makes it really hard to inspect the diff... can you keep this function immediately following `func ExtractList` so the diff is obvious?

Reply:
ok~ I have re-adjusted the position, I hope to be able to help you.